app/controllers/application_controller.rb

```diff
@@ -459,15 +459,15 @@ def process_params_options(params)
                 end
     end
 
-    if params[:model_id] && !params[:active_tree]
-      curr_model_id = from_cid(params[:model_id])
-      unless curr_model_id.nil?
-        options[:parent] = identify_record(curr_model_id, controller_to_model) if curr_model_id && options[:parent].nil?
+    if params[:id] && !params[:active_tree]
+      parent_id = from_cid(params[:parent_id])
+      unless parent_id.nil?
+        options[:parent] = identify_record(parent_id, controller_to_model) if parrent_id && options[:parent].nil?
       end
     end
 
     options[:parent] = options[:parent] || @parent
-    options[:association] = HAS_ASSOCATION[params[:model]] if HAS_ASSOCATION.include?(params[:model])
+    options[:association] = HAS_ASSOCATION[params[:model_name]] if HAS_ASSOCATION.include?(params[:model_name])
```

Comment:
I am a bit confused by mixing :association and :model name, those were 2 different things before?

Reply:
e.g before if you had controller_name/1/?diplay=all_vms

it could map to this 
https://github.com/Ladas/manageiq-ui-classic/blob/36c5a957b6fcad6a89d81bb8238c70ef04a55d1e/app/controllers/mixins/generic_show_mixin.rb#L151

or going to generic, which does https://github.com/Ladas/manageiq-ui-classic/blob/36c5a957b6fcad6a89d81bb8238c70ef04a55d1e/app/controllers/mixins/generic_show_mixin.rb#L139

while expecting to have list of allowed sub-tables per controller like this https://github.com/Ladas/manageiq-ui-classic/blob/35dbd795ba08c3778866245dfce91b4b5d88a76d/app/controllers/network_port_controller.rb#L18

---
of course the explorer views were more complex and weren't ported to this format :-(

Reply:
so to sum what I try to say :-)

- we had allowed list of controller subtables
- we were catching specific sublists in generic_show_mixins, because they could not be generic for some reason (we are not passing all args to UI and back to report data now)
- naming is confusing (but it was confusing before) since get_view takes model as a first arg which is taken from @display , so maybe we need to take this up to @lpichler :-)

Reply:
> of course the explorer views were more complex and weren't ported to this format :-(
￼ 
Yes, the explorer views where/are even more mess. I already limited the number of different patterns for nested lists in explorer view but it's still pretty messy. It combines the single and list views deeper in the code and it will need more work to cleanup.

Naming is and was confusing the methods on the get_view* code path have different names for the same thing on several places and it's not easy to fix.

Let's do small steps that improve the situation. We don't need to fix too much in a single PR.